### PR TITLE
Components - De-hardcoded the UI metadata file path in GCP components

### PR DESCRIPTION
**Pull request by Ark-kun**
Part of the effort to de-hardcode all output paths in the components which is problematie for portability.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/2697)

### CODE REVIEW

File: components/gcp/dataflow/launch_python/component.yaml [hunk 1]
```diff
@@ -61,10 +61,10 @@ implementation:
       --staging_dir, {inputValue: staging_dir},
       --requirements_file_path, {inputValue: requirements_file_path},
       --args, {inputValue: args},
-      --wait_interval, {inputValue: wait_interval}
+      --wait_interval, {inputValue: wait_interval},
+      --ui-metadata-path, {outputPath: MLPipeline UI metadata},
```
Comment (numerology) [inline components/gcp/dataflow/launch_python/component.yaml]: nit: is there any specific reason why ui-metadata-path follows k8s naming while other args follows pythonic naming?

Comment (Ark-kun) [reply components/gcp/dataflow/launch_python/component.yaml]: I'd say that it follows Linux command-line parameter scheme. http://man7.org/linux/man-pages/man1/sort.1.html http://man7.org/linux/man-pages/man1/git.1.html https://docs.python.org/3/library/argparse.html#dest

I'd say that dashes are more natural for command-line, but I can change them to underscores if you want.

Comment (numerology) [reply components/gcp/dataflow/launch_python/component.yaml]: Got it. I'm fine either way.

**Pull request merged**